botfront/imports/ui/components/settings/ImportRasaFiles.jsx

```diff
@@ -16,46 +13,100 @@ import {
-import { getDefaultDomainAndLanguage } from '../../../lib/story.utils';
-import {
-    useStoryFileReader,
-    useDomainFileReader,
-    useDatasetFileReader,
-} from './fileReaders';
-import {
-    handleImportStoryGroups,
-    handleImportDomain,
-    handleImportDataset,
-} from './fileImporters';
+import { useMutation } from '@apollo/react-hooks';
+import { useFileReader } from './fileReaders';
 import { ProjectContext } from '../../layouts/context';
+import { unZipFile } from '../../../lib/importers/common';
+import { importFilesMutation } from './graphql';
 
-const ImportRasaFiles = (props) => {
-    const {
-        existingStoryGroups, existingSlots, projectId, defaultDomain,
-    } = props;
-    const { projectLanguages, instance, language } = useContext(ProjectContext);
-    const [fallbackImportLanguage, setFallbackImportLanguage] = useState(language);
-    const [wipeCurrent, setWipeCurrent] = useState({
-        stories: false,
-        domain: false,
-        'NLU data': false,
-    });
+const ImportRasaFiles = () => {
+    const { projectLanguages, project: { _id: projectId }, language } = useContext(ProjectContext);
+    const [importFiles] = useMutation(importFilesMutation);
+    const [fallbackImportLanguage, setFallbackImportLanguage] = useState();
+    const [importResults, setImportResults] = useState([]);
+    useEffect(() => setFallbackImportLanguage(language), [language]);
+    const [wipeCurrent, setWipeCurrent] = useState(false);
+    const [importSummary, setImportSummary] = useState([]);
+
+    const [filesImporting, setFilesImporting] = useState(false);
+
+    const validateFunction = async (files) => {
+        // we don't want to send files with errors already so we filter those out
+        // but we keep their index so it easy to recontruct the list
+        const filesNotSentIndexes = [];
+        const filesToSend = files.filter((file, index) => {
+            if (file.errors && file.errors.length > 0) {
+                filesNotSentIndexes.push(index);
+                return false;
+            }
+            return true;
+        });
+        const validationResult = await importFiles({
+            variables: {
+                projectId,
+                files: filesToSend,
+                onlyValidate: true,
+                wipeCurrent,
+                fallbackLang: fallbackImportLanguage,
+            },
+        });
+        const validationData = validationResult?.data?.import?.fileMessages;
+        const summary = validationResult?.data?.import?.summary;
+        setImportSummary(summary);
+        if (validationData.length !== files.length) {
+            // that means some files were not sent
+            filesNotSentIndexes.forEach((index) => {
+                validationData.splice(index, 0, files[index]); // so we re-insert those
+            });
+        }
+        return validationData;
+    };
+
+    const handleImport = async ([files, setFileList]) => {
```

Comment:
There is one, that happen when we drop the file https://github.com/botfront/botfront/blob/3d5f0a7a93fb16f360d14f1e720763fa280bf25b/botfront/imports/ui/components/settings/fileReaders.js#L51-L57
However, I agree it could be improved with a size limit. 50 Mb per file sounds fine to me , what do you think ?

Reply:
alright for 50MB (not Mb), is there a limit to the overall number of files ?

Reply:
there is no limit to the overall number of files, I'm not sure that would be really interesting to put a hard "limit" that blocks importing when there is too much file. imo, A warning saying that the import might take some time with lots of file would be better as it won't stop an user from importing